test/extended/router/gatewayapicontroller.go

```diff
@@ -539,3 +591,74 @@ func assertHttpRouteSuccessful(oc *exutil.CLI, gwName, name string) (*gatewayapi
 	e2e.Logf("httpRoute %s/%s successful", namespace, name)
 	return checkHttpRoute, nil
 }
+
+// assertHttpRouteConnection checks if the http route of the given name replies successfully,
+// and returns an error if not
+func assertHttpRouteConnection(oc *exutil.CLI, hostname, gwName string) error {
+	gwapiClient := gatewayapiclientset.NewForConfigOrDie(oc.AdminConfig())
+	ingressNameSpace := "openshift-ingress"
+	gateway, errGwStatus := gwapiClient.GatewayV1().Gateways(ingressNameSpace).Get(context.TODO(), gwName, metav1.GetOptions{})
+	if errGwStatus != nil || gateway == nil {
+		e2e.Failf("Unable to create httpRoute, no gateway available during route assertion %v", errGwStatus)
+	}
+
+	// Create the http client to check the header.
+	client := &http.Client{
+		Timeout: 10 * time.Second,
+		Transport: &http.Transport{
+			TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
+		},
+	}
+
+	// Make sure the DNSRecord is ready to use.
+	assertDNSRecordStatus(oc, gwName, "", "")
```

Comment:
Can you please move it in the test (`Ensure HTTPRoute object is created`), after the creation of the gateway? This way we will separate the concerns, this function should do one thing - check the connection to a route.

Reply:
done